Framework/Kernel/src/ErrorReporter.cpp

```diff
@@ -0,0 +1,141 @@
+                             Types::Core::time_duration upTime,
+                             std::string exitCode, bool share, std::string name,
+                             std::string email)
+    : m_application(application), m_exitCode(exitCode), m_upTime(upTime),
+      m_share(share), m_name(name), m_email(email) {
+  int retval = Mantid::Kernel::ConfigService::Instance().getValue(
+      "errorreports.rooturl", m_url);
+  if (retval == 0) {
+    g_log.error() << "Failed to load error report url\n";
+  }
+}
+
+/** Generates an error message and then calls an internet helper to send it
+*/
+void ErrorReporter::sendErrorReport() {
+  try {
+    std::string message = this->generateErrorMessage();
+
+    // send the report
+    // Poco::ActiveResult<int> result = m_errorActiveMethod(message);
+    this->sendReport(message, m_url + "/api/error");
+  } catch (std::exception &ex) {
+    g_log.debug() << "Send error report failure. " << ex.what() << '\n';
+  }
+}
+
+/** Generates an error message in json format
+*/
+std::string ErrorReporter::generateErrorMessage() {
+  ::Json::Value message;
+
+  // username
+  message["uid"] = Kernel::ChecksumHelper::md5FromString(
+      ConfigService::Instance().getUsername());
+  // hostname
+  message["host"] = Kernel::ChecksumHelper::md5FromString(
+      ConfigService::Instance().getComputerName());
+
+  // os name, version, and architecture
+  message["osName"] = ConfigService::Instance().getOSName();
+  message["osArch"] = ConfigService::Instance().getOSArchitecture();
+  message["osVersion"] = ConfigService::Instance().getOSVersion();
+  message["osReadable"] = ConfigService::Instance().getOSVersionReadable();
+
+#if defined(MAKE_VATES)
+  // paraview
+  message["ParaView"] = Kernel::ParaViewVersion::targetVersion();
+#else
+  message["ParaView"] = 0;
+#endif
+
+  // mantid version and sha1
+  message["mantidVersion"] = MantidVersion::version();
+  message["mantidSha1"] = MantidVersion::revisionFull();
+
+  message["dateTime"] =
+      Types::Core::DateAndTime::getCurrentTime().toISO8601String();
+
+  message["upTime"] = to_simple_string(m_upTime);
+
+  message["application"] = m_application;
+
+  message["facility"] = ConfigService::Instance().getFacility().name();
+
+  message["exitCode"] = m_exitCode;
+
+  if (m_share) {
+    message["email"] = m_email;
+    message["name"] = m_name;
+  }
+
+  ::Json::FastWriter writer;
+  return writer.write(message);
+}
+
+/** Submits a post request to the specified url with the message as the body
+ @param message : String containg json formatted error message
+ @param url : The url to send the post request to
+*/
+int ErrorReporter::sendReport(const std::string &message,
+                              const std::string &url) {
+  int status = -1;
+  try {
+    Kernel::InternetHelper helper;
+    std::stringstream responseStream;
+    helper.setTimeout(20);
+    helper.setBody(message);
+    status = helper.sendRequest(url, responseStream);
```

Comment:
Maybe a `.debug` log message to indicate the status and response for future debugging?

Reply:
There is already some debug level information about the post request within Internet helper with responds with for example,

Sending http POST request to: http://errorreports.mantidproject.org/api/error
Answer from web: 201 Created
Continue working.

Do we need more in addition to this ?

Reply:
No that's okay. I'd forgotten about the other level of logging.